ci/check-documentation.py

```diff
@@ -39,12 +39,44 @@ def find_md_files():
 def main():
     all_passed = True
 
+    index_file_paths = (
+        OMZ_ROOT / 'models/intel/index.md',
+        OMZ_ROOT / 'models/public/index.md',
+        OMZ_ROOT / 'demos/README.md',
+    )
+
+    all_md_files = tuple(find_md_files())
+
     def complain(message):
         nonlocal all_passed
         all_passed = False
         print(message, file=sys.stderr)
 
-    for md_path in sorted(find_md_files()):
+    index_child_md_links = {}
+    for check_case_path in index_file_paths:
+        if not check_case_path.exists():
+            complain(f'{check_case_path}: file not found')
+            continue
+
+        required_md_links = []
+        for md_file in all_md_files:
+            if md_file.name == "README.md":
+                try:
+                    md_rel_path = md_file.relative_to(check_case_path.parent)
+                except ValueError:
+                    continue
+
+                md_intermediate_parents = list(md_rel_path.parents)[1:-1] # removed root and first parent dirs
+
+                if not any((check_case_path.parent / parent_dir / 'README.md').exists()
+                for parent_dir in md_intermediate_parents) and md_file != check_case_path:
```

Comment:
```suggestion
                        for parent_dir in md_intermediate_parents) and md_file != check_case_path:
```

I would also suggest to move the second half of the condition to the enclosing `if` and rephrase it as `md_file.parent != check_case_path.parent`. That way:

* we don't just exclude the index file itself, but also any (hypothetical) README file in the same directory;
* someone who reads the code can be certain that `md_rel_path.parents` has at least two components, so they don't have to wonder if `[1:-1]` slice works as intended. :-)